schemes/sima_diagnostics/check_energy_gmean_diagnostics.F90

```diff
@@ -16,43 +16,53 @@ module check_energy_gmean_diagnostics
    private
    save
 
+   public :: check_energy_gmean_diagnostics_init
    public :: check_energy_gmean_diagnostics_run  ! main routine
 
-CONTAINS
+  ! Private module options.
+  logical :: print_global_means = .false.
+
+contains
+
+!> \section arg_table_check_energy_gmean_diagnostics_init Argument Table
+!! \htmlinclude arg_table_check_energy_gmean_diagnostics_init.html
+  subroutine check_energy_gmean_diagnostics_init(print_global_means_in)
+    ! Input arguments
+    logical,            intent(in)    :: print_global_means_in
+
+    print_global_means = print_global_means_in
+  end subroutine check_energy_gmean_diagnostics_init
 
    !> \section arg_table_check_energy_gmean_diagnostics_run  Argument Table
    !! \htmlinclude check_energy_gmean_diagnostics_run.html
    subroutine check_energy_gmean_diagnostics_run( &
+      amIRoot, &
+      iulog, &
       teinp_glob, &
       teout_glob, &
       heat_glob,  &
       psurf_glob, &
       ptopb_glob, &
       errmsg, errflg)
 
-      use spmd_utils,   only: masterproc
-      use cam_logfile,  only: iulog
-      use time_manager, only: get_nstep
+      logical,            intent(in)  :: amIRoot
+      integer,            intent(in)  :: iulog         ! log output unit
 
-      !------------------------------------------------
-      !   Input / output parameters
-      !------------------------------------------------
       real(kind_phys),    intent(in)  :: teinp_glob    ! global mean energy of input state [J m-2]
       real(kind_phys),    intent(in)  :: teout_glob    ! global mean energy of output state [J m-2]
       real(kind_phys),    intent(in)  :: psurf_glob    ! global mean surface pressure [Pa]
       real(kind_phys),    intent(in)  :: ptopb_glob    ! global mean top boundary pressure [Pa]
```

Comment:
Should we remove these variables if they are no longer being used below?

Reply:
Thanks, removed!